master/buildbot/data/builds.py

```diff
@@ -53,7 +54,13 @@ def get(self, resultSpec, kwargs):
             bldr = kwargs['builderid']
             num = kwargs['number']
             dbdict = yield self.master.db.builds.getBuildByNumber(bldr, num)
-        defer.returnValue((yield self.db2data(dbdict))
+
+        try:
+            props = yield self.master.db.builds.getBuildProperties(dbdict['id'])
+        except (TypeError, KeyError):
+            props = {}
+
+        defer.returnValue((yield self.db2data(dbdict, props))
                           if dbdict else None)
```

Comment:
Is it useful to check dbdict? At this step, dbdict must be valid (not None) else it will crash ln 59 when requesting for "id" key, isn't it?